src/code_checker.bzl

```diff
@@ -310,6 +313,8 @@ def _code_checker_impl(ctx):
                 compilation_context = target[CcInfo].compilation_context
                 for src in srcs:
                     args = target[CompileInfo].arguments[src]
+                    if src.path.endswith(header_extensions):
+                        continue
```

Comment:
However, this change is a bit questionable and needs to be checked: basically srcs should not contain header files, but if if does should we skip analysis for them?